vm/src/object/core.rs

```diff
@@ -717,6 +717,19 @@ impl PyObject {
         }
     }
 
+    pub fn delete_dict(&self) -> Result<(), ()> {
```

Comment:
set_dict in core.rs is only used for object_set_dict. Rather than adding delete_dict, I prefer to add this logic into `set_dict` if other problems not found.